style/react.md

```diff
@@ -74,29 +39,128 @@ return (
 );
 ```
 
-80 chars is a bit tight so we opt to conserve the extra 4.
-
-## Align properties
-
-Fit them all on the same line if you can, but put them all in the same column if you have to break.
-
-Good:
-
-`<div className="highlight" key="highlight-div">`
+#### Align properties.
 
-Good
+Fit them all on the same line if you can, but put them all in the same
+column if you have to break.  This makes it easy to see the props at a
+glance.
 
+Yes:
 ```jsx
+<div className="highlight" key="highlight-div">
 <div
     className="highlight"
     key="highlight-div">
 ```
 
-Bad
-
+No:
 ```jsx
 <div className="highlight"
-    key="highlight-div"
+     key="highlight-div"
 ```
 
-This makes it easy to see the props at a glance.
+#### Only export a single react class.
+
+Every .jsx file should export a single react class, and nothing else.
+This is for testability: the fixture framework requires it to
+function.
+
+Note the file can still define multiple classes, it just can't export
+more than one.
+
+
+---------------------
+### Language features
+
+#### Prefer [props to state](http://facebook.github.io/react/docs/interactivity-and-dynamic-uis.html).
+
+You almost always want to use props.  Copying data from props to state
+can cause the UI to get out of sync from the underlying data models
+because what's displayed no longer directly tracks what's stored.  By
+having the data in one place instead of copying it, you ensure
+everything is consistent.
+
+#### Use [propTypes](http://facebook.github.io/react/docs/reusable-components.html).
+
+React Components should always have complete propTypes.  Every
+attribute of this.props should have a corresponding entry in
+propTypes.  This documents that props need to be passed to a model.
+([example](http://jsfiddle.net/spicyj/DEpwb/))
```

Comment:
Maybe link to an example from our own code? https://github.com/Khan/webapp/blob/32aa862769d4e93c477dc0ee0388816056252c4a/javascript/search-package/search-results-list.jsx#L14